src/main/java/org/spongepowered/api/profile/GameProfile.java

```diff
@@ -37,9 +44,40 @@
 public interface GameProfile extends Identifiable, DataSerializable {
 
     /**
-     * Gets the player's last known username.
+     * Creates a {@link GameProfile} from the provided ID and name.
+     *
+     * @param uniqueId The unique ID
+     * @param name The name
+     * @return The created profile
+     */
+    static GameProfile of(UUID uniqueId, @Nullable String name) {
+        return Sponge.getServer().getGameProfileManager().createProfile(uniqueId, name);
+    }
+
+    /**
+     * Gets the name associated with this profile.
+     *
+     * @return The associated name if present, otherwise {@link Optional#empty()}
+     */
+    Optional<String> getName();
+
+    /**
+     * Gets the property map for this profile.
+     *
+     * @return The property map
+     */
+    ProfilePropertyMap getPropertyMap();
```

Comment:
You should clarify that this is a live map (unless you're going to add a setter method).